onnxruntime/core/providers/nnapi/nnapi_builtin/builders/impl/leakyrelu_op_builder.cc

```diff
@@ -0,0 +1,146 @@
+
+using namespace onnxruntime::nnapi::op_builder_helpers;
+
+class LeakyReluOpBuilder : public BaseOpBuilder {
+  // Add operator related
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const NodeUnit& node_unit) const override;
+
+  // Operator support related
+ private:
+  bool IsOpSupportedImpl(const InitializedTensorSet& initializers, const NodeUnit& node_unit,
+                         const OpSupportCheckParams& params) const override;
+
+  // LeakyRelu opset 6- has unsupported attributes
+  int GetMinSupportedOpSet(const NodeUnit& /* node_unit */) const override { return 6; }
+};
+
+// Add operator related
+
+Status LeakyReluOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
+                                                 const NodeUnit& node_unit) const {
+  auto& shaper(model_builder.GetShaper());
+  const auto& operand_types(model_builder.GetOperandTypes());
+  const auto& operand_indices(model_builder.GetOperandIndices());
+
+  const auto& input = node_unit.Inputs()[0].node_arg.Name();
+  const auto& output = node_unit.Outputs()[0].node_arg.Name();
+
+  auto input_shape = shaper[input];
+
+  NodeAttrHelper helper(node_unit);
+  const auto alpha = helper.Get("alpha", 0.01f);
+
+  // We will use multiple operations to simulate LeakyRelu here, including NNAPI ANEURALNETWORKS_SELECT/
+  // ANEURALNETWORKS_LESS/ANEURALNETWORKS_MUL.
+  // input X = [-1, 0, 1]
+  // multiply by the alpha value: Z = alpha * X = [-0.1, 0, 0.1]
+  // then construct the boolean input X less than zero: C = [false, true, true], true means selecting element from the first input,
+  // and false vice versa.
+  // output Y = [-0.1, 0, 1]
+  InlinedVector<uint32_t> input_indices;
+  input_indices.push_back(operand_indices.at(input));
+
+  // Add Less operation - Less(X, Zero)
+  int count = std::accumulate(input_shape.begin(), input_shape.end(), 1, std::multiplies<int>());
+  std::vector<float> zero_vec(count, 0.0f);
+
+  std::string zero_vec_name = model_builder.GetUniqueName(node_unit.Name() + input + "_zero_cast");
+  std::string less_output_name = model_builder.GetUniqueName(node_unit.Name() + input + "_less_than_zero");
+
+  const OperandType zero_operand_type(Type::TENSOR_FLOAT32, input_shape);
+  ORT_RETURN_IF_ERROR(model_builder.AddOperandFromPersistMemoryBuffer(zero_vec_name, zero_vec.data(),
+                                                                      zero_operand_type));
+  input_indices.push_back(operand_indices.at(zero_vec_name));
+
+  OperandType less_output_operand_type(Type::TENSOR_BOOL8, input_shape);
+  shaper.AddShape(less_output_name, input_shape);
+  ORT_RETURN_IF_ERROR(model_builder.AddOperation(ANEURALNETWORKS_LESS,
+                                                 input_indices, {less_output_name}, {less_output_operand_type}));
+
+  // Add Mul operation - Mul(Alpha, X)
+  input_indices.clear();
+  std::vector<float> alpha_vec(count);
+  std::fill(alpha_vec.begin(), alpha_vec.end(), alpha);
```

Comment:
`std::vector<float> alpha_vec(count, alpha);`?

But do we need to create a vector? Looks like NNAPI LESS and MUL support broadcasting so a single value for the zero and alpha instead of a vector should work.